safe_eth/eth/clients/etherscan_client_v2.py

```diff
@@ -0,0 +1,56 @@
+import os
+from typing import Any, Dict, List, Optional
+from urllib.parse import urljoin
+
+import requests
+
+from safe_eth.eth import EthereumNetwork
+from safe_eth.eth.clients import EtherscanClient
+
+
+class EtherscanClientV2(EtherscanClient):
+    BASE_API_V2_URL = "https://api.etherscan.io"
+
+    def __init__(
+        self,
+        network: EthereumNetwork,
+        api_key: Optional[str] = None,
+        request_timeout: int = int(
+            os.environ.get("ETHERSCAN_CLIENT_REQUEST_TIMEOUT", 10)
+        ),
+    ):
+        super().__init__(EthereumNetwork.MAINNET, api_key, request_timeout)
+        self.network = network
+        self.base_api_url = self.BASE_API_V2_URL
+
+    @classmethod
+    def _get_supported_networks(cls) -> List[Dict[str, Any]]:
+        """
+        Fetches a list of supported networks by the Etherscan API v2.
+
+        :return: List of supported networks, or empty list if request fails.
+        """
+        url = urljoin(cls.BASE_API_V2_URL, "v2/chainlist")
+        response = requests.get(url)
+        if response.ok:
+            return response.json().get("result", [])
+        return []
+
+    @classmethod
+    def is_supported_network(cls, network: EthereumNetwork) -> bool:
+        """
+        Checks if a given Ethereum network is supported by the Etherscan API v2.
+
+        :param network: The Ethereum network to check.
+        :return: True if the network is supported; False otherwise.
+        """
+        supported_networks = cls._get_supported_networks()
+        return any(
+            item.get("chainid") == str(network.value) for item in supported_networks
+        )
+
+    def build_url(self, path: str) -> str:
```

Comment:
This method should be at the top of the class, just for clarity

Reply:
Done [here](https://github.com/safe-global/safe-eth-py/pull/1407/commits/9b8049300f80f04aa75dd4b1f643d0735ab3d0ed)